Replace nginx with haproxy

andrewgdotcom opened the pull request.

andgal-siren committed:
Replace nginx with haproxy

andgal-siren committed:
Improved expiry on stick-table

andgal-siren committed:
int32 limit

andgal-siren committed:
Improved documentation niggles

mclueppers committed:
CHORE: Update HAProxy configuration template

Add configuration flags to enable trust of Cloudflare and X-Forward-For headers.

mclueppers committed:
CHORE: Update HAProxy configuration template

Add configuration flags to enable trust of Cloudflare and X-Forward-For headers.

mclueppers committed:
Merge branch 'chore-haproxy-improvements' of github.com:mclueppers/hockeypuck into chore-haproxy-improvements

andgal-siren committed:
Merge pull request #19 from mclueppers/chore-haproxy-improvements

CHORE: Update HAProxy configuration template

mclueppers committed:
CHORE: Track client IP from txn.clientIP and/or CF-Connecting-IP

Now that we have a predictable and trusted source of truth what the client IP is,
use txn.clientIP varible in src ACL rules, for example is_tor_exit_relay.
Refactor the configuration file and remove code duplication as well as improve the
DDoS protection logic to ban repeat offenders early enough.

andgal-siren committed:
QAD method of soft reloading haproxy every 30m, assumes haproxy has predictable PID inside container (FRAGILE AS HELL)

andgal-siren committed:
More parameterisation

andgal-siren committed:
Custom haproxy entrypoint using ps from alpine to identify correct process

andgal-siren committed:
Merge pull request #20 from mclueppers/chore-haproxy-improvements

Track client IP from txn.clientIP and/or CF-Connecting-IP

andgal-siren committed:
Better haproxy maintenance guidance

andgal-siren committed:
added find-keys.bash

andgal-siren committed:
find-keys can now read patterns from stdin

mclueppers committed:
CHORE: Add haproxy-internal container to persist stick-table data

Public HAProxy instance is being infrequently restarted to update SSL certs or
Tor exit relays lists. By adding an internal only HAProxy instance we can keep
stick-tables in sync and hence persist them.
Next to it, change the stick-tables type from `ip` to `ipv6`.

mclueppers commented on contrib/docker-compose/standalone/haproxy/etc/haproxy.cfg.tmpl:
```diff
@@ -0,0 +1,490 @@
+
+    # Set CF-Connecting-IP header to the tracked client IP. All subsequent requests from this client will have this header set.
+    # And all ACL rules in backends etc. must be evaluated against this header.
+    http-request set-header CF-Connecting-IP var(txn.clientIP)
```
TODO: expand to `%[var(txn.clientIP)]`

andgal-siren committed:
delete-keys.bash can now read fingprints from a file

mclueppers committed:
CHORE: Fix URL redirect for Prometheus

Fix URL Location redirects from Prometheus. Revert if we do
https://github.com/prometheus/prometheus/issues/4925

andgal-siren committed:
Merge pull request #22 from mclueppers/chore-fix-haproxy-prometheus

CHORE: Fix URL redirect for Prometheus

andgal-siren committed:
Merge pull request #21 from mclueppers/chore-haproxy-persist

CHORE: Add haproxy-internal container to persist stick-table data

andgal-siren committed:
use SQLCMD -c

andgal-siren committed:
Merge remote-tracking branch 'origin/haproxy-new' into haproxy-new

andgal-siren committed:
Fix haproxy_internal unix socket

andgal-siren committed:
Fix race condition in certbot; improved instructions

andgal-siren committed:
Remove foreign key constraint on bulk deletion

andgal-siren committed:
Expose haproxy peering port to localhost but allow override

andgal-siren committed:
Parameterise all external host:port configs

andgal-siren committed:
Allow letsencrypt directory override (e.g. for sharing with host OS)

andgal-siren committed:
Fix missing % expansion

andgal-siren committed:
use docker-compose and .env in SQLCMD

andrewgdotcom committed:
Haproxy deconstruction (#23)

Explode haproxy config into haproxy.d/*; use live envars not envsubst; create alias.map

andrewgdotcom committed:
Haproxy configuration improvements

* Pass envars more elegantly
* Write advanced HAProxy config options to .env, commented out
* Enable cloudflare/proxy by envar
* move redundant check outside loop
* Fix ipv6 support

andrewgdotcom committed:
Haproxy comment out configs using envar definitions (#25)

andgal-siren committed:
Remove redundant context from notices

andgal-siren committed:
Improved README.md

andgal-siren committed:
Add shim configuration

andgal-siren committed:
remove some howlers

andrewgdotcom committed:
Instructions and tooling to help build clusters (#27)

andgal-siren committed:
revert typo

sebix committed:
Configuration parameters for log format and prometheus (#26)

* make log format configurable

* add switch to disable prometheus

---------

Co-authored-by: Sebastian Wagner <swagner@intevation.de>

andgal-siren committed:
Add migration 1a

andgal-siren committed:
fix howler

andgal-siren committed:
Tidy up docker-compose

andgal-siren committed:
bump docker-compose version for consistency

andgal-siren committed:
Improved documentation

andgal-siren committed:
Fix prometheus

andgal-siren committed:
Update README.md for prometheus IP whitelisting

andgal-siren committed:
Deny poliely when ip not in prometheus whitelist

andrewgdotcom committed:
Haproxy mksite/mkconfig simplification (#28)

* Migrate haproxy config file housekeeping from mkconfig.bash to entrypoint

* Move docker-compose specific envar defs into docker-compose.yml file

andgal-siren committed:
fix metrics_path for custom web.external-url

andgal-siren committed:
Fix up shim mode to match recent changes in normal mode; fix regression in migration 3

andgal-siren committed:
Remove redundant instruction step

andgal-siren committed:
Fix shell metacharacter parsing issue

andgal-siren committed:
more shell metacharacter parsing

andgal-siren committed:
safer .env file handling in scripts; not all systems have base32, use tr instead; many double-quote warnings

andgal-siren committed:
fix regression that replaced 30 bytes of entropy with 30 base32 chars

andgal-siren committed:
Remove redundant quotes

andgal-siren committed:
Shim mode no longer includes prometheus

andgal-siren committed:
Update README.md for recent changes

andrewgdotcom commented:
Closing this in favour of a PR upstream. cc @mclueppers :-)
